pilot/pkg/xds/eds.go

```diff
@@ -132,6 +133,7 @@ func (s *DiscoveryServer) edsCacheUpdate(clusterID, hostname string, namespace s
 		// unnecessary full push which can become a real problem if a pod is in crashloop and thus endpoints
 		// flip flopping between 1 and 0.
 		s.deleteEndpointShards(clusterID, hostname, namespace)
+		s.updateServiceAccount(hostname, namespace)
```

Comment:
For eds cache update, we already do the service account update right? It is only not done for service delete?

Reply:
That cal is not complete, it only cal the passed in IstioEndpoints